src/mscorlib/shared/System/UnitySerializationHolder.cs

```diff
@@ -0,0 +1,311 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.Serialization;
+using System.Reflection;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+
+namespace System
+{
+    // Holds classes (Empty, Null, Missing) for which we guarantee that there is only ever one instance of.
+#if CORECLR
+    internal
+#else
+    public  // On CoreRT, this must be public because of the Reflection.Core/CoreLib divide and the need to whitelist past the ReflectionBlock.
+#endif
+    class UnitySerializationHolder : ISerializable, IObjectReference
+    {
+        internal const int EmptyUnity = 0x0001;
```

Comment:
We don't use any of these save NullUnity, and don't plan to use them either. If we remove all the others, a bunch of other code could be removed from this file, and it would still be compatible?

Reply:
I would do that, would cut this file in half

Reply:
Agreed, though it begs the question as to whether any have any short-term benefits in including, because of dependencies elsewhere that are blocked similarly to how the DBNull case blocks some. (Any longer term benefits can be decided on later).